hook.go

```diff
@@ -135,11 +137,23 @@ func (h *SumoLogicHook) queue(msg SumoLogicMesssage) (err error) {
 }
 
 func (h *SumoLogicHook) upload(b []byte) error {
-	payload := [][]byte{b}
+	payload := bytes.Join([][]byte{b}, newline)
+
+	// gzip the bytes to be uploaded
+	var buf bytes.Buffer
+	g := gzip.NewWriter(&buf)
+	if _, err := g.Write(payload); err != nil {
+		return err
```

Comment:
I think it will be good to log error so user is aware what went wrong with uploading log
`h.errorf("compressing  payload - %s", err)`